### PR TITLE
feat: modify user agent string

### PR DESCRIPTION
**Pull request by sbansla**
# Fixes #

A short description of what this PR does.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-java/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: src/main/java/com/twilio/http/HttpUtility.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import com.twilio.Twilio;
+import lombok.experimental.UtilityClass;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@UtilityClass
+class HttpUtility {
+    String getUserAgentString(final List<String>... userAgentExtensions) {
```
Comment (shrutiburman) [inline src/main/java/com/twilio/http/HttpUtility.java:11]: Can we add a dedicated test case for this util ?

Comment (sbansla) [reply src/main/java/com/twilio/http/HttpUtility.java:11]: I have covered in NetworkHttpClientTest (method: testGetUserAgentExtensible). As this utility is being called from httpclient implementations.
Should I write a separate file for that ?

File: src/main/java/com/twilio/Twilio.java [hunk 2]
```diff
@@ -71,6 +73,35 @@ public static synchronized void init(final String username, final String passwor
...
+     * @param userAgentExtensions offers ability to extend the User-Agent string
+     */
+    public static synchronized void init(final String username, final String password, final String accountSid,
+                                         List<String> userAgentExtensions) {
+        Twilio.setUsername(username);
+        Twilio.setPassword(password);
+        Twilio.setAccountSid(accountSid);
+        Twilio.setUserAgentExtensions(userAgentExtensions);
+    }
+
```
Comment (childish-sambino) [inline src/main/java/com/twilio/Twilio.java:104]: Prefer to not overload `init` any more. Customers can simply call `setUserAgentExtensions` instead.

File: src/main/java/com/twilio/http/HttpUtility.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        StringBuilder userAgentString = new StringBuilder();
+        userAgentString.append("twilio-java/")
+                .append(Twilio.VERSION)
+                .append(" (")
+                .append(Twilio.OS_NAME)
+                .append(Twilio.OS_ARCH)
```
Comment (childish-sambino) [inline src/main/java/com/twilio/http/HttpUtility.java:17]: Should there be a separator between these 2?

File: src/main/java/com/twilio/http/HttpUtility.java (hunk 1 above)
Comment (childish-sambino) [inline src/main/java/com/twilio/http/HttpUtility.java:11]: I don't think we should accept a vararg as input. Simpler to just accept a single `List`.

File: src/main/java/com/twilio/http/NetworkHttpClient.java [hunk 4]
```diff
@@ -30,17 +30,17 @@ public class NetworkHttpClient extends HttpClient {
     /**
      * Create a new HTTP Client.
      */
-    public NetworkHttpClient() {
-        this(DEFAULT_REQUEST_CONFIG);
+    public NetworkHttpClient(final List<String>... userAgentExtensions) {
+        this(DEFAULT_REQUEST_CONFIG, userAgentExtensions);
```
Comment (childish-sambino) [inline src/main/java/com/twilio/http/NetworkHttpClient.java:34]: Again, not in favor of the additional overloading, but understand the issues with building the headers as part of the constructor. Is there a path where the extensions are set/added _after_ construction and then included in when calling `makeRequest`?

**Comment by childish-sambino** [comment]
Should remove the ticket number from the PR title since the title will be included in the public changelog but internal tickets are not public.

File: src/main/java/com/twilio/Twilio.java (hunk 2 above)
Comment (sbansla) [reply src/main/java/com/twilio/Twilio.java]: addressed

File: src/main/java/com/twilio/http/HttpUtility.java (hunk 3 above)
Comment (sbansla) [reply src/main/java/com/twilio/http/HttpUtility.java]: addressed. Added a space.

File: src/main/java/com/twilio/http/HttpUtility.java (hunk 1 above)
Comment (sbansla) [reply src/main/java/com/twilio/http/HttpUtility.java]: addressed. I have removed varargs completely from the code.

File: src/main/java/com/twilio/http/NetworkHttpClient.java (hunk 4 above)
Comment (sbansla) [reply src/main/java/com/twilio/http/NetworkHttpClient.java]: addressed, within makeRequest we can add header
(I have tested that as well by creating a dummy service and calling that service by adding header similar to makeRequest.)

**Comment by sbansla** [comment]
> Should remove the ticket number from the PR title since the title will be included in the public changelog but internal tickets are not public.

I have renamed the branch. I have create a different PR for that branch

**Pull request closed without merging**